B9PartSwitch/UI/UIPartActionSubtypeSelector.cs

```diff
@@ -111,6 +111,21 @@ public override void Setup(UIPartActionWindow window, Part part, PartModule part
             subtypeTitleText.text = switcherModule.CurrentSubtype.title;
 
             subtypeButtons[currentButtonIndex].Activate();
+
+            // up to 7 subtypeButtons fit in the PAW without scrolling
+            if (subtypeButtons.Count > 7)
+            {
+                scrollMain.scrollSensitivity = subtypeButtons.Count;
```

Comment:
Have you tested that this is still usable with very long lists?

Reply:
I've tested up to 104 subtypes.  I could try making a part with 1k or 10k just to see what happens.  This is one of those cases where any value is necessarily making a tradeoff between speed and precision.  For me, being able to quickly get to the general region of the list I wanted to be in and then clicking or dragging to refine felt best.  That's part of why I keep centering the current subtype's button, so there's always three options ahead and behind.  But I'm not opposed to capping the value or something.

Reply:
I did some more testing to see how many ticks of the mouse wheel it takes to get from the start to the end of the subtype list.
- 32 subtypes:	22 ticks
- 104 subtypes:	26 ticks
- 1040 subtypes:	28 ticks
- 10400 subtypes:	28 ticks

I definitely recommend _against_ anyone putting 10k subtypes on a part, but it does work... eventually.

Reply:
Dang wow.  I guess I was more concerned about like, if you want something in the middle of a very long list how easy is it to get there without scrolling past it?  I think tens of subtypes is probably reasonable for testing though.  Any more and we have to think of a new method of finding them.

Reply:
From my testing, roughly 150 to 200 subtypes is the transition zone.  Below that, each tick of the mouse wheel moves the list fewer than 7 buttons, so no button should ever be skipped past.  E.g., with 31 subtypes, it takes 6 ticks of the mouse wheel to scroll a button from one edge of the viewport off the opposite edge.

Feel is more challenging because there's significant variation in mouse wheels.  A sensitivity that feels right for a loose, relatively free-spinning wheel will feel sluggish on a tight, clicky wheel.  I also wouldn't be surprised if there are other sources of variation.  On the positive side, it's easy for us to add a sensitivity scale factor if the current setting is too high/low for most users.  Making it user-configurable is probably overkill, but it's always there as a last resort if we can't find a good one-size-fits-all setting.  Have you had a chance to try it out yourself?  What kind of mouse do you use?  My mouse wheel is relatively tight (if spun, it stops itself within a few ticks), so I'm probably biased towards higher sensitivity.

Reply:
Alright that's satisfying enough for me.